### PR TITLE
Cleanup ClinicalDataType in study view api's

### PR DESCRIPTION
**Pull request by kalletlak**
# What? Why?
Part of fixing https://github.com/cBioPortal/cbioportal/issues/4983

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -56,11 +56,9 @@ import {
     generateScatterPlotDownloadData,
     getChartMetaDataType,
     getClinicalAttributeUniqueKey,
-    getClinicalAttributeUniqueKeyByDataTypeAttrId,
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: Let's directly remove this method to make sure all places are removed.

Comment (zhx828) [reply src/pages/studyView/StudyViewPageStore.ts]: We can use ClinicalAttributeID as the uniqueKey going forward, but let's put that logic in the `getClinicalAttributeUniqueKey` so we can switch easily next time if the requirement changes

File: src/pages/studyView/StudyViewPageStore.ts [hunk 2]
```diff
@@ -1645,7 +1635,7 @@ export class StudyViewPageStore {
         onError: (error => {}),
         onResult: (data=>{
             data.forEach(item => {
-                const uniqueKey = getClinicalAttributeUniqueKeyByDataTypeAttrId(item.clinicalDataType, item.attributeId);
+                const uniqueKey = item.attributeId;
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: Let's use the method getClinicalAttributeUniqueKey

File: src/pages/studyView/StudyViewPageStore.ts [hunk 3]
```diff
@@ -1843,11 +1828,11 @@ export class StudyViewPageStore {
         });
 
         _.forEach(sampleClinicalData, item => {
-            clinicalData[item.uniqueSampleKey] = { ...(clinicalData[item.uniqueSampleKey] || {}), ['SAMPLE_' + item.clinicalAttributeId]: item.value };
+            clinicalData[item.uniqueSampleKey] = { ...(clinicalData[item.uniqueSampleKey] || {}), [item.clinicalAttributeId]: item.value };
         })
 
         let patientClinicalData = await defaultClient.fetchClinicalDataUsingPOST({
-            'clinicalDataType': "PATIENT",
+            'clinicalDataType': ClinicalDataTypeEnum.PATIENT,
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: Thanks for updating all these strings

File: src/pages/studyView/StudyViewPageStore.ts [hunk 4]
```diff
@@ -3655,8 +3637,7 @@ export class StudyViewPageStore {
         let sampleClinicalDataMap = await this.getClinicalDataBySamples(this.selectedSamples.result)
 
         let clinicalAttributesNameSet = _.reduce(this.clinicalAttributes.result, (acc, next) => {
-            let id = (next.patientAttribute ? 'PATIENT' : 'SAMPLE') + '_' + next.clinicalAttributeId;
-            acc[id] = next.displayName;
+            acc[next.clinicalAttributeId] = next.displayName;
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: getClinicalAttributeUniqueKey

File: src/pages/studyView/StudyViewPageStore.ts (hunk 2 above)
Comment (kalletlak) [reply src/pages/studyView/StudyViewPageStore.ts]: getClinicalAttributeUniqueKey accepts ClinicalAttribute, which is not possible here

File: src/pages/studyView/StudyViewPageStore.ts (hunk 1 above)
Comment (kalletlak) [reply src/pages/studyView/StudyViewPageStore.ts]: its already removed

**Comment by zhx828** [comment]
I don't think at the moment, we need to worry about much of the e2e unless it's relevant. @kalletlak

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 5]
```diff
@@ -1606,7 +1606,7 @@ export class ResultsViewPageStore {
         // Add meta information for each of the clinical attribute
         // Convert to a Set for easy access and to update attribute meta information(would be useful while adding new features)
         _.reduce(this.clinicalAttributes.result, (acc: { [id: string]: ChartMeta }, attribute) => {
-            const uniqueKey = getClinicalAttributeUniqueKey(attribute);
+            const uniqueKey = getUniqueKey(attribute);
```
Comment (alisman) [inline src/pages/resultsView/ResultsViewPageStore.ts]: why make the name of this so generic?

**Pull request merged**